app/autocomplete_mention/controller.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+var database = require('../../lib/mongo/');
+
+module.exports.mention = function mention(req,res) {
+  database.connect(function(err,db){
+    if(err) throw err;
+    var query = req.query.pattern;
+    db.collection('user').find({
+      name : {$regex:query}
```

Comment:
I tried this approach, but `$text` doesn't match partial words the way `$regex` does.  
e.g., 
after setting ```> db.user.createIndex( { name:'text' } )```,
```> db.user.find({ $text : {$search:'wo'} })``` returns nothing, while 
```> db.user.find({ $text : {$search:'work'} })``` returns
```{ "_id" : ObjectId("55d546114b270fc1712aaec9"), "name" : "work" }```.
Since we are searching specifically for a partial match, `$text` doesn't seem right for the job.
From mongo's docs: 
> The $text operator matches on the complete stemmed word. So if a document field contains the word blueberry, a search on the term blue will not match. However, blueberry or blueberries will match.